src/app/component/eco-news/components/news-list/news-list.component.ts

```diff
@@ -30,20 +31,39 @@ export class NewsListComponent implements OnInit, OnDestroy {
     private ecoNewsSelectors: EcoNewsSelectors) { }
 
   ngOnInit() {
+    this.onResize();
     this.setNullList();
     this.checkUserSingIn();
     this.userOwnAuthService.getDataFromLocalStorage();
   }
 
-  private checkUserSingIn(): void {
-    this.userOwnAuthService.credentialDataSubject
-      .subscribe((data) => this.isLoggedIn = data && data.userId);
+  public onResize(): void {
+    this.windowSize = window.innerWidth;
+    this.view = (this.windowSize > 576) ? this.view : true;
   }
 
   public onScroll(): void {
     this.addElemsToCurrentList();
   }
 
+  public changeView(event: boolean): void {
+    this.view = event;
+  }
+
+  public getFilterData(value: Array<string>): void {
+    if (this.gridOutput !== value) {
+      this.setNullList();
+      this.gridOutput = value;
+    }
+    this.addElemsToCurrentList();
+  }
+
+  private checkUserSingIn(): void {
+    this.userOwnAuthService.credentialDataSubject
+      .subscribe((data) => this.isLoggedIn = data && data.userId);
+  }
+
+
   private addElemsToCurrentList(): void {
     this.ecoNewsSubscription = this.gridOutput ?
       this.ecoNewsSubscription = this.ecoNewsService.getNewsListByTags(this.iterator++, 12, this.gridOutput)
```

Comment:
assignment sentence inside a ternary operator - can be rewrite more correct

Reply:
it wasn`t my code, but I fixed it.